server/java/src/main/java/com/godaddy/asherah/grpc/AppEncryptionConfig.java

```diff
@@ -63,7 +66,37 @@ KeyManagementService setupKeyManagementService(final String kmsType, final Strin
 
       case Constants.METASTORE_DYNAMODB:
         logger.info("using DynamoDB-based metastore...");
-        return DynamoDbMetastoreImpl.newBuilder().build();
+        DynamoDbMetastoreImpl.Builder builder = DynamoDbMetastoreImpl.newBuilder();
+        String dynamoDbRegion = dynamoDbConfig.getDynamoDbRegion();
+        String keySuffix = dynamoDbConfig.getKeySuffix();
+        String tableName = dynamoDbConfig.getTableName();
+        String endPoint = dynamoDbConfig.getDynamoDbEndpointConfig();
+        String signingRegion = dynamoDbConfig.getDynamoDbSigningRegion();
+
+        // Check for region configuration
+        if (!StringUtils.isEmpty(dynamoDbRegion)) {
+          builder.withRegion(dynamoDbRegion);
+        }
+        // Check for endpoint configuration
+        if (endPoint != null || signingRegion != null) {
+          if (!StringUtils.isEmpty(endPoint) && !StringUtils.isEmpty(signingRegion)) {
+            builder.withEndPointConfiguration(endPoint, signingRegion);
+          }
+          else {
+            logger.error("One or more parameter(s) for endpoint configuration missing.");
+            return null;
+          }
+        }
+        //Check for table name
+        if (!StringUtils.isEmpty(tableName)) {
+          builder.withTableName(tableName);
+        }
+        // Check for key suffix
+        if (!StringUtils.isEmpty(keySuffix)) {
```

Comment:
I am still able to start the server with `--enable-key-suffix ' '` and `--enable-key-suffix ''`
This should be failing.
Be good to add a corresponding test also for the same.

Reply:
`--enable-key-suffix ' '` is interpreted as a string with 2 single quotes, thereby failing the requirement for a null or an empty string. This would be a user error and I don't think we need to worry about that here. you shouldn't be able to start with server with `--key-suffix ""`
Have added a test case that checks that null is returned for empty strings